data/generate_ember_average_intensities_co2.js

```diff
@@ -0,0 +1,68 @@
+const fs = require("fs");
+const csv = fs.readFileSync("data/co2-intensities-ember-2021.csv");
+const getHeaders = require("./utils/getCSVHeaders");
+const mapCountries = require("./utils/mapCountries");
+const getCountryCodes = require("./utils/getCountryCodes");
+
+const countries = mapCountries();
+const array = csv.toString().split("\n");
+
+/* Store the converted result into an array */
+const csvToJsonResult = {};
+const gridIntensityResults = {}
+
+/* Store the CSV column headers into seprate variable */
+const headers = getHeaders(array[0]);
+
+/* Iterate over the remaining data rows */
+for (let currentArrayString of array) {
+	// If there's an empty line, keep calm and carry on.
+	// Also, skip the first row since those are the headers.
+	if (currentArrayString.length === 0 || currentArrayString === array[0]) continue;
+
+	/* Empty object to store result in key value pair */
+	const jsonObject = {}
+	/* Store the current array element */
+	let string = ''
+	let quoteFlag = 0
+	for (let character of currentArrayString) {
+		if (character === '"' && quoteFlag === 0) {
+			quoteFlag = 1
+		} else if (character === '"' && quoteFlag == 1) quoteFlag = 0
+		if (character === ',' && quoteFlag === 0) character = '|'
+		if (character !== '"') string += character
+	}
+
+	let jsonProperties = string.split("|")
+	let country = jsonProperties[0] || jsonProperties[1];
+
+	for (let j in headers) {
```

Comment:
would you mind using a longer variable name than `J` here?

it was a bit hard to see what `j` was referring to further down